### ISSUE TITLE
How to import xml wiki?

### ISSUE THREAD

**Issue by peddy22** [gnosygnu/xowa#700]
I downloaded English Wikipedia in the xml format (file enwiki-202003-pages-articles-multistream.xml) and hit Tools->Import Offline in xowa. After some time, the progress messages seem to have stopped. I cannot find where to access the wiki however. Can someone give me a hand? Thanks!

**Comment by bpfeffer** [comment]
Mine did that last week, where it acted like it started and created a lock file; I had better luck using the "generate script" and then "run script" buttons after I selected the proper drop-down items. That gave good feedback during the creation, including going from 00.00% and went through building files for me.

The file-picker doesn't work in "read from file" for me so I had to just paste the full path to the file itself in the file location box.

I also put the categorylinks, page_props, and pagelinks files in the same folder as the pages-articles file and it I think imported them and processed them all at the same time when it built it, as it deleted the files and my search and categories seem to work in the created database.

**Comment by peddy22** [comment]
Thanks for the reply, I just had to wait a bit longer. I did get warnings for the categorylinks and pagelinks sql files as well, even though they had been downloaded. Seems to be working fine.

**Comment by ndxf** [comment]
as far as I understand, the 17GB download only includes text for english wikiepdia. It doesn't include images so xowa app downloads the images on the fly from wikipedia as you open articles in xowa.

**Comment by gnosygnu** [comment]
Hey, sorry about the late response, but I stopped getting notifications on my gmail account.

Responding to points above:

> I downloaded English Wikipedia in the xml format (file enwiki-202003-pages-articles-multistream.xml) and hit Tools->Import Offline in xowa

XOWA uses the `page-articles.xml` file. I'm surprised it didn't barf completely with `pages-articles-multistream.xml`, but the format is probably similar. The link for 2020-03 is here: https://dumps.wikimedia.org/enwiki/20200301/enwiki-20200301-pages-articles.xml.bz2

> The file-picker doesn't work in "read from file" for me so I had to just paste the full path to the file itself in the file location box.

That stinks. Out of curiosity, are you on a Mac / Linux box? If so what version is your OS?

> I also put the categorylinks, page_props, and pagelinks files in the same folder as the pages-articles file and it I think imported them and processed them all at the same time when it built it, as it deleted the files and my search and categories seem to work in the created database.

Yup, that's the right way to do it! I think I have that documented somewhere.

> as far as I understand, the 17GB download only includes text for english wikiepdia. It doesn't include images so xowa app downloads the images on the fly from wikipedia as you open articles in xowa.

Yup, `Import Offline` and `Import Online` is only for the wikitext (no images). To get images, you can go to https://archive.org/download/Xowa_enwiki_2020-02 and download all the files starting with `Xowa_enwiki_2020-02_file`. When done, unzip them and place the `.xowa` files in the same directory as your existing enwiki `.xowa` files and you'll have images.

Let me know if you run into issues. I'll check the issues manually.

Thanks!

**Comment by bpfeffer** [comment]
> ...
> > The file-picker doesn't work in "read from file" for me so I had to just paste the full path to the file itself in the file location box.
> 
> That stinks. Out of curiosity, are you on a Mac / Linux box? If so what version is your OS?
>
Xowa is installed on a network share and I'm running Xowa on a CentOS 6 server across Xserver to my CentOS 7 workstation.  I can see directories but no files, and the file extension picker dropdown isn't selectable for me. I'm one or two releases behind, so this could have been fixed since.
 
> > I also put the categorylinks, page_props, and pagelinks files in the same folder as the pages-articles file and it I think imported them and processed them all at the same time when it built it, as it deleted the files and my search and categories seem to work in the created database.
> 
> Yup, that's the right way to do it! I think I have that documented somewhere.

I didn't see anything explicit about having them all together, I just was trying to use the "wiki maintenance" menu to do the stages of category links and fixing searching, but this method of leaving all of the files in one folder and it doing it automatically. I'll definitely do it this way in the future!

**Comment by gnosygnu** [comment]
>  I can see directories but no files, and the file extension picker dropdown isn't selectable for me. I'm one or two releases behind, so this could have been fixed since.

Nah, I haven't made changes in recent releases. I'll take a look at it on my openSUSE box later. If not reproducible there, I'll try on a CentOS distro. 

> I just was trying to use the "wiki maintenance" menu to do the stages of category links and fixing searching, but this method of leaving all of the files in one folder and it doing it automatically.

Cool. Nice deduction on your side. I'll add it to documentation later. Thanks!
